### PR TITLE
Adding new course

### PR DESCRIPTION
**Pull request by Ashu104**
## What does this PR do?
Add resource(s)

### Description
new free course on graph theory is added

### Why is this valuable (or not)?
design and taught by MIT professor.

### How do we know it's really free?
content freely available on MIT's OCW. 

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Review by 3R1Dev — APPROVED** [review]
Everything looks good, thanks for the  PR

**Comment by Ashu104** [comment]
hi @3R1Dev .
I'm glad for your response. yah. its been nice to be part of this amazing project.

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -730,6 +730,7 @@
 
 * [Graph Theory](https://youtube.com/playlist?list=PLztBpqftvzxXBhbYxoaZJmnZF6AUQr1mH) - Wrath of Math
 * [Graph Theory - DM](https://youtube.com/playlist?list=PLYrahs7hsYIQiSNxTfZndQz7jWPXsA1ur) - SCCI Labs IIT Ropar
+* [Graph Theory And Additive Combinatorics](https://ocw.mit.edu/courses/18-217-graph-theory-and-additive-combinatorics-fall-2019/video_galleries/video-lectures/) -Prof. Yufei Zhao (MIT)
```
Comment (eshellman) [inline courses/free-courses-en.md:733]:
```suggestion
* [Graph Theory And Additive Combinatorics](https://ocw.mit.edu/courses/18-217-graph-theory-and-additive-combinatorics-fall-2019/video_galleries/video-lectures/) - Prof. Yufei Zhao (MIT)
```
good addition!

File: more/free-programming-cheatsheets.md [hunk 2]
```diff
@@ -116,6 +116,8 @@
 ### Data Science
 
 * [Cheatsheets for Data Scientists](https://www.datacamp.com/community/data-science-cheatsheets) - Datacamp (PDF)
+* [Data Science Cheat Sheets](https://medium.com/analytics-vidhya/data-science-cheat-sheets-109ddcb1aca8) - Surya Gutta (HTML, PDF)
```
Comment (eshellman) [inline more/free-programming-cheatsheets.md:119]: This is a blog post that embeds 3 cheatsheets from other sites. We prefer to link directly (and give credit to) to the original sources.

File: more/free-programming-cheatsheets.md [hunk 3]
```diff
@@ -116,6 +116,8 @@
...
+* [Data Science Cheatsheets](https://goodjob0823.github.io/Data-Science-Cheat-Sheets) - goodjob0823 (HTML, PDF)
```
Comment (eshellman) [inline more/free-programming-cheatsheets.md:120]: this is a list of links to cheat sheets. Our list is not a list of lists. romove this.

File: more/free-programming-cheatsheets.md [hunk 4]
```diff
@@ -294,7 +296,9 @@
 
 ### Machine Learning
 
+* [Cheat Sheet of Machine Learning and Python (and Math) Cheat Sheets](https://medium.com/machine-learning-in-practice/cheat-sheet-of-machine-learning-and-python-and-math-cheat-sheets-a4afe4e791b6) - Robbie Allen (HTML)
```
Comment (eshellman) [inline more/free-programming-cheatsheets.md:299]: this is another list. remove it.

File: more/free-programming-cheatsheets.md [hunk 5]
```diff
@@ -294,7 +296,9 @@
...
 * [Machine Learning Cheat Sheet](https://www.datacamp.com/cheat-sheet/machine-learning-cheat-sheet) - DataCamp Team (HTML)
+* [Machine Learning Glossary](https://ml-cheatsheet.readthedocs.io/en/latest) - ml-cheatsheet (HTML)
```
Comment (eshellman) [inline more/free-programming-cheatsheets.md:301]:
```suggestion
* [Machine Learning Glossary](https://ml-cheatsheet.readthedocs.io/en/latest) - Brendan Fortuner
```

**Review by eshellman — CHANGES_REQUESTED** [review]

**Comment by Ashu104** [comment]
@3R1Dev 
can you please consider the changes I've made after your suggestions.

File: courses/free-courses-en.md [hunk 6]
```diff
@@ -730,6 +730,7 @@
 
 * [Graph Theory](https://youtube.com/playlist?list=PLztBpqftvzxXBhbYxoaZJmnZF6AUQr1mH) - Wrath of Math
 * [Graph Theory - DM](https://youtube.com/playlist?list=PLYrahs7hsYIQiSNxTfZndQz7jWPXsA1ur) - SCCI Labs IIT Ropar
+* [Graph Theory And Additive Combinatorics](https://ocw.mit.edu/courses/18-217-graph-theory-and-additive-combinatorics-fall-2019/video_galleries/video-lectures/) - Prof. Yufei Zhao (MIT)
```
Comment (eshellman) [inline courses/free-courses-en.md:733]: we don't list honorifics such as "Prof."

**Review by eshellman — CHANGES_REQUESTED** [review]

**Comment by eshellman** [comment]
it seems you are not using a fresh branch for new PRs. As a result, new problems are introduced after the reviews! Please rebase to the version  that has been reviewed.

**Pull request closed without merging**